### ISSUE TITLE
Workspace Support

### ISSUE THREAD

**Issue by sxzz** [rolldown/tsdown#44]

**Comment by RSS1102** [comment]
Using [pnpm](https://pnpm.io/) seems to support monorepo very quickly?

**Comment by sxzz** [comment]
Since finding and loading `tsdown.config.*` is a time-consuming operation, I envision consolidating the configurations scattered across various packages into a single file in the form of a configuration file to speed up the overall packaging process.

Currently, it is also possible to use pnpm workspace to simultaneously launch multiple tsdown CLI instances.

**Comment by RSS1102** [comment]
> Since finding and loading `tsdown.config.*` is a time-consuming operation, I envision consolidating the configurations scattered across various packages into a single file in the form of a configuration file to speed up the overall packaging process.

I was just about to ask if it's possible to load `tsdown.config.*` files.😂😂 Maybe should have an example, or create cli.

---
> Currently, it is also possible to use pnpm workspace to simultaneously launch multiple tsdown CLI instances.
why have **launch multiple tsdown CLI instances** ，I don't quite understand this~

---

But it seems that it needs to support many plug-ins, so is it a better idea to transform it into a monorepo now?

**Comment by sxzz** [comment]
tsdown is still in the early stages of the project, please refer to the usage documentation of [tsup](https://github.com/egoist/tsup). Some options of tsup are not yet implemented.

I don't quite understand your question about monorepo. Of course you can just use one tsdown instance if that meets your needs.

**Comment by RSS1102** [comment]
ok. 
sorry, I thought it was tsdown => monorepo

**Comment by fu050409** [comment]
@sxzz I'm not sure how you envision this feature. Maybe it's returning an array of multiple configuration files when `defineConfig` is called, and setting `root` in each configuration object? I think this may take more overheads on resolving and validating each config.

But apart from that, I have an idea. Could we allow configuring `workspace.members` like Cargo does, and enable setting default options for each subproject in the workspace in `workspace.package`?

**Comment by sxzz** [comment]
Correct, there are two approaches—centralized or decentralized. I believe having a root config that specifies which packages should be included (similar to `pnpm-workspace.yaml`) and inheriting configurations from the root would be ideal.

**Comment by fu050409** [comment]
@sxzz I am interested in this, would you mind if I make some attempts at this? I can try to resolve this issue if I can.

**Comment by sxzz** [comment]
Sure, please go ahead
